### PR TITLE
Update performance benchmark for reproducibility

### LINKED ISSUE #52: DOC: Include instructions on how to access the datasets used for benchmarks
To ensure reproducibility, instructions should be added on how to access datasets exclusively on LLNL resources, as with this example

https://github.com/xCDAT/xcdat-validation/blob/main/scripts/performance-benchmarks/perf_benchmark.py

This prevents full reproducibility of the results

Related to https://github.com/openjournals/joss-reviews/issues/6426

### PR DESCRIPTION
**Pull request by tomvothecoder**
Closes #52 

- [x] Add `README.md` with instructions
- [x] Add ESGF wget scripts for datasets
- [x] Update filenames for scripts to include suffix (1_, 2_)
- [x] Add `01_esgf_download_datasets.py` to run all ESGF wget scripts and store outputs in `input-datasets`
  - [x] Check if the files are available on Globus -- added links to the `README.md`
- [x] Add `02_create_cdms2_xmls.py` to create cdscan XMLs 
- [x] Perform an end-to-end test run using `README.md` instructions

### CODE REVIEW

File: scripts/performance-benchmarks/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Performance Benchmark
```
Comment (tomvothecoder) [inline scripts/performance-benchmarks/README.md:1]: This `README.md` has all of the instructions.

**Review by tomvothecoder — COMMENTED** [review]
Hey @mgrover1, can you checkout this branch (`joss-paper-review`) and try following  the instructions in `scripts/performance-benchmarks/README.md`?

I recommend using the ESGF2 Globus endpoint to download the data because it is fast. I also provided a Python script (`1_esgf_download_datasets.py`) to download the datasets using ESGF wget scripts, but it is much slower than Globus.

**Comment by mgrover1** [comment]
@tomvothecoder - I can confirm the benchmarks + process here! Thank you!

File: scripts/performance-benchmarks/README.md (hunk 1 above)
Comment (mgrover1) [reply scripts/performance-benchmarks/README.md:1]: Thanks!!

**Comment by tomvothecoder** [comment]
@mgrover1 Thanks Max! I will merge this PR

File: scripts/performance-benchmarks/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+#### Specifications for original machine for performance benchmarks for JOSS paper (Dec/2023)
+
+- OS: RHEL 7
+- Memory: 1,000 GiB
+- CPU: Intel(R) Xeon(r) CPU E7-8890v4 @ 2.20GHz
+
+#### How xCDAT is configured for parallelism
+
+- Uses Dask Distributed local sechduler with multiprocessing
```
Comment (tomvothecoder) [inline scripts/performance-benchmarks/README.md:111]:
```suggestion
- Uses Dask Distributed local scheduler with multiprocessing
```

**Pull request merged**